autofill/autofill-store/src/main/java/com/duckduckgo/autofill/store/urlmatcher/AutofillDomainNameUrlMatcher.kt

```diff
@@ -27,17 +27,19 @@ import timber.log.Timber
 class AutofillDomainNameUrlMatcher @Inject constructor() : AutofillUrlMatcher {
 
     override fun extractUrlPartsForAutofill(originalUrl: String?): ExtractedUrlParts {
-        if (originalUrl == null) return ExtractedUrlParts(null, null)
+        if (originalUrl == null) return unextractable()
 
         val normalizedUrl = originalUrl.normalizeScheme()
         return try {
-            val eTldPlus1 = normalizedUrl.toHttpUrl().topPrivateDomain()
+            val httpUrl = normalizedUrl.toHttpUrl()
+            val eTldPlus1 = httpUrl.topPrivateDomain()
             val domain = originalUrl.extractDomain()
+            val port = httpUrl.port
```

Comment:
Not sure about the consequences of this but getting the port from `HttpUrl` will always returns a port. If this is only used for matching then it's fine but if at some points it leaks into title, etc we might show ports like 80. Getting the port from URI will return -1 if there is no port and it won't infer 80, etc.

Reply:
I think it's fine here because it is just used for matching, and elsewhere for display (currently `cleanRawUrl`) it handles it differently to omit _implicit_ ports.